hack/coco-cm-defaulter.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: 'cm-defaulter'
+  namespace: openshift-sandboxed-containers-operator
+spec:
+  ttlSecondsAfterFinished: 100
+  completions: 1
+  parallelism: 1
+  template:
+    spec:
+      hostNetwork: true # needed for imds access # TODO: comment out
+      securityContext:
+        runAsUser: 0 # needed for container mode dnf access
+
+      restartPolicy: Never
+      containers:
+      - command:
+        - 'bash'
+        - '-c'
+        - |
+          dnf install jq -y
+          CLOUD_PROVIDER=$(oc get infrastructure -n cluster -o json | jq '.items[].status.platformStatus.type'  | awk '{print tolower($0)}' | tr -d '"' )
+          echo "CLOUD_PROVIDER=${CLOUD_PROVIDER}" > /tmp/cm.env
+
+          if [ "${CLOUD_PROVIDER}" = "azure" ]; then
+              AZURE_NSG_ID=${AZURE_NSG_ID}
+              [[ ! "${AZURE_REGION}" ]] && AZURE_REGION=$(curl -H Metadata:true --noproxy "*" "http://169.254.169.254/metadata/instance/compute/location?api-version=2017-08-01&format=text")
+              [[ ! "${AZURE_RESOURCE_GROUP}" ]] && AZURE_RESOURCE_GROUP=$(curl -H Metadata:true --noproxy "*" "http://169.254.169.254/metadata/instance/compute/resourceGroupName?api-version=2017-08-01&format=text")
+              [[ ! "${AZURE_SUBSCRIPTION_ID}" ]] && AZURE_SUBSCRIPTION_ID=$(curl -H Metadata:true --noproxy "*" "http://169.254.169.254/metadata/instance/compute/subscriptionId?api-version=2017-08-01&format=text")
+              [[ ! "${AZURE_SUBNET_ID}" ]] && AZURE_SUBNET_ID="/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP}/providers/Microsoft.Network/virtualNetworks/${AZURE_RESOURCE_GROUP%-rg}-vnet/subnets/${AZURE_RESOURCE_GROUP%-rg}-worker-subnet"
+              [[ ! "${AZURE_NSG_ID}" ]] && AZURE_NSG_ID="/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP}/providers/Microsoft.Network/networkSecurityGroups/${AZURE_RESOURCE_GROUP%-rg}-nsg"
+
+          cat <<EOF >> /tmp/cm.env
+              AZURE_IMAGE_ID="*****"
+              AZURE_NSG_ID=${AZURE_NSG_ID}
+              AZURE_REGION=${AZURE_REGION}
+              AZURE_RESOURCE_GROUP=${AZURE_RESOURCE_GROUP}
+              AZURE_SUBNET_ID=${AZURE_SUBNET_ID}
+              AZURE_SUBSCRIPTION_ID=${AZURE_SUBSCRIPTION_ID}
+              AZURE_INSTANCE_SIZE=Standard_DC2as_v5
+              DISABLE_CLOUD_CONFIG=true
+              DISABLECVM=false
+              PROXY_TIMEOUT=5m
+              VXLAN_PORT=9000
+          EOF
+          else
+            echo "Uknown provider: \"${CLOUD_PROVIDER}\""
+          fi
+
+           cat /tmp/cm.env
+           oc create cm peer-pods-cm --from-env-file=/tmp/cm.env
+
+        image: 'registry.redhat.io/openshift4/ose-cli'
```

Comment:
Is there an alternate public image available? I think pulling from registry.redhat.io needs auth, which is fine. But if there is a public equivalent it's better.

Reply:
quay.io/openshift/origin-cli ?
It's uses oc 4.2 though

Reply:
Or you'll need to download the cli from a public url like - https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.13.19/
Would it help to use a regular fedora image and use the oc from the above link ? or it's simply over engineering ;)

Reply:
I think for generic usage we anyway need to adapt few things. And since this is specific for OCP, we can go with the the existing image as-is